src/art/art.c

```diff
@@ -308,6 +324,43 @@ static inline art_indexed_child_t art_node4_lower_bound(
     return indexed_child;
 }
 
+static bool art_internal_validate_at(const art_node_t *node,
+                                     art_internal_validate_t validator);
+
+static bool art_node4_internal_validate(const art_node4_t *node,
+                                        art_internal_validate_t validator) {
+    if (node->count == 0) {
+        return art_validate_fail(&validator, "Node4 has no children");
+    }
+    if (node->count > 4) {
+        return art_validate_fail(&validator, "Node4 has too many children");
+    }
+    validator.depth++;
+    if (node->count == 1 && !IS_LEAF(node->children[0])) {
```

Comment:
Even if it is a leaf, it should also be combined.

Reply:
Right, don't know what I was thinking here.